packages/react/src/side-nav/SideNavLinkList.tsx

```diff
@@ -1,51 +1,110 @@
-import { useMemo } from 'react';
-import { SideNavGroup } from './SideNavGroup';
-import { SideNavLink } from './SideNavLink';
+import { useCallback, useMemo } from 'react';
+import { SideNavUnorderedList } from './SideNavUnorderedList';
+import { SideNavListItem } from './SideNavListItem';
 import { SideNavProps } from './SideNav';
 import { hasNestedActiveItem } from './utils';
+import { SideNavLink } from './SideNavLink';
 
 export type SideNavLinkListProps = {
 	activePath: string | undefined;
 	items: SideNavProps['items'];
+	nestedItemsVariant: SideNavProps['nestedItemsVariant'];
+};
+
+type ItemWithIsActive = Omit<SideNavLinkListProps['items'][number], 'items'> & {
+	isActive: boolean;
+	items?: ItemWithIsActive[];
 };
 
-export function SideNavLinkList({
+const addIsActive =
+	(activePath: SideNavLinkListProps['activePath']) =>
+	(
+		item: SideNavProps['items'][number] | ItemWithIsActive
+	): ItemWithIsActive => ({
+		...item,
+		isActive:
+			item.href === activePath || hasNestedActiveItem(item.items, activePath),
+		items: item.items?.length
```

Comment:
- [ ] If there are nested active items, update the `items` otherwise leave them.
- [ ] Any non-active branches should not recursively check/add `isActive`.